FreeRTOS/coreHTTP
# Add Ability to Stop Parsing After HTTP Headers

Add Ability to Stop Parsing After HTTP Headers

Description
-----------
This pull request adds the ability to stop parsing immediately after HTTP headers. This feature is beneficial for handling chunked encoding scenarios. Prior to this update, llhttp would automatically begin parsing the body, which lead to issues with tracking chunks and their respective sizes in cases where buffer capacity is exceeded.
This PR addresses the issue by introducing `HTTP_RESPONSE_DO_NOT_PARSE_BODY_FLAG`, which halts parsing after the headers and stores any remaining data in the body buffer. The subsequent processing is intended to be handled by a higher-level library.

Test Steps
-----------
With the implementation of patches from #167, processing responses becomes possible even when the initial buffer capacity is exceeded. However, without the HTTP_RESPONSE_DO_NOT_PARSE_BODY_FLAG, processing chunked responses in this manner is not achievable.

Checklist:
----------

Related Issue
-----------
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
